src/routes/popup/receive.tsx

```diff
@@ -1,105 +1,133 @@
-import { Button, Section, Tooltip } from "@arconnect/components";
-import { CopyIcon, ShareIcon } from "@iconicicons/react";
+import {
+  ButtonV2,
+  Section,
+  Tooltip,
+  useToasts,
+  type DisplayTheme
+} from "@arconnect/components";
 import { useStorage } from "@plasmohq/storage/hook";
 import { ExtensionStorage } from "~utils/storage";
+import { CheckIcon, CopyIcon } from "@iconicicons/react";
 import { formatAddress } from "~utils/format";
 import { QRCodeSVG } from "qrcode.react";
 import browser from "webextension-polyfill";
-import Head from "~components/popup/Head";
 import styled from "styled-components";
 import copy from "copy-to-clipboard";
-import { useEffect } from "react";
+import { useEffect, type MouseEventHandler, useState } from "react";
 import { PageType, trackPage } from "~utils/analytics";
 import HeadV2 from "~components/popup/HeadV2";
+import { useTheme } from "~utils/theme";
 
 export default function Receive() {
   // active address
   const [activeAddress] = useStorage<string>({
     key: "active_address",
     instance: ExtensionStorage
   });
+  const [copied, setCopied] = useState(false);
 
   //segment
   useEffect(() => {
     trackPage(PageType.RECEIVE);
   }, []);
 
+  const theme = useTheme();
+  const { setToast } = useToasts();
+
+  const copyAddress: MouseEventHandler = (e) => {
+    e.stopPropagation();
+    copy(activeAddress);
+    setCopied(true);
+    setTimeout(() => setCopied(false), 1000);
+    setToast({
+      type: "success",
+      duration: 2000,
+      content: `${formatAddress(activeAddress, 3)} ${browser.i18n.getMessage(
+        "copied_address_2"
+      )}`
+    });
+  };
+
   return (
     <Wrapper>
       <div>
         <HeadV2 title={browser.i18n.getMessage("receive")} />
-        <AddressField>
-          {formatAddress(activeAddress ?? "", 6)}
-          <Tooltip
-            content={browser.i18n.getMessage("copy_address")}
-            position="bottom"
-          >
-            <CopyButton onClick={() => copy(activeAddress)} />
-          </Tooltip>
-        </AddressField>
       </div>
-      <Section>
-        <QRCodeWrapper>
-          <QRCodeSVG
-            fgColor="#fff"
-            bgColor="transparent"
-            size={256}
-            value={activeAddress ?? ""}
-          />
-        </QRCodeWrapper>
-      </Section>
-      <Section>
-        <Button fullWidth>
-          {browser.i18n.getMessage("share")}
-          <ShareIcon />
-        </Button>
-      </Section>
+      <ContentWrapper>
+        <Section style={{ paddingBottom: "8px" }}>
+          <QRCodeWrapper displayTheme={theme}>
+            <QRCodeSVG
+              fgColor="#fff"
+              bgColor="transparent"
+              size={285.84}
+              value={activeAddress ?? ""}
+            />
+          </QRCodeWrapper>
+        </Section>
+        <Section style={{ paddingTop: "8px" }}>
```

Comment:
This big button doesn't copy the address. We should copy the address and show the toast if we click in any place within this button